Source/Blazorise/Components/ModalProvider/ModalProvider.razor.cs

```diff
@@ -0,0 +1,149 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Blazorise.Modules;
+using Blazorise.States;
+using Blazorise.Utilities;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+#endregion
+
+namespace Blazorise
+{
+
+    /// <summary>
+    /// A modal provider to be set at the root of your app, providing a programmatic way to invoke modals with custom content by using ModalService.
+    /// </summary>
+    public partial class ModalProvider : BaseComponent
+    {
+        #region Members
+
+        private List<ModalInstance> modalInstances;
```

Comment:
Might be better to use `Stack<>`?

Reply:
Actually I don't recall if I setup the API, but the Show returns the ModalInstance, so the user could use the service to close it, so there should be a lookup to close a certain modal instance. In that case the List would make more sense in my opinion. I'll look it up, also see if it even works properly (Closing other instances)

Reply:
Yep, added an Hide(modalInstance), my guess is that it won't have much use, but it's there if some use case comes up.
So I guess we should keep the List?

Reply:
I don't see a scenario where the user could close the previous modal. of always goes from the front visible to the back.

Reply:
UHm there's always someone ingenious haha...
maybe some warning modal that generates another modal and then it can make sense to remove the warning modal sooner...
Think we should just remove that API, go with the stack?

Reply:
Yeah, I guess the warning modal would mess things up if it shows behind the currently open modal. OK in that case we'll stick with the List.